src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,162 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> first;
+    private Node<T> last;
+
+    public MyLinkedList() {
+        first = new Node<>(null, null, null);
+        last = new Node<>(null, null, null);
+    }
+
+    private void checkIndexException(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException();
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        Node<T> newElem;
+        if (first.item == null) {
+            newElem = new Node<>(null, value, last);
+            first = newElem;
+            last.prev = first;
+        } else if (last.item == null) {
```

Comment:
@Korfat Do you need this?? Why you want to add at the end? Can't we add one by one starting from the first?